### ISSUE TITLE
continuous-atomic images hang in cloud-final.service

### ISSUE THREAD

**Issue by martinpitt** [cockpit-project/cockpit#7432]
I noticed that test runs on verify/continuous-atomic take more than a day. https://fedorapeople.org/groups/cockpit/logs/pull-7423-20170802-053333-a42f046b-verify-continuous-atomic/log.html was retried this morning after the previous run (I don't have the URL any more) hung since yesterday.

Using this issue to keep notes and comparing hung states between runs.

Running processes for the above run:
```
$ virsh list
 Id    Name                           State
----------------------------------------------------
 192   continuous-atomic-127.0.0.2-2901 running
 250   continuous-atomic-127.0.0.2-2201 running

sh-4.3$ ps ux|cat
USER        PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
user          1  0.0  0.0  11788  1592 ?        Ss   04:14   0:00 /bin/sh /usr/local/bin/cockpit-tests --publish=sink --verbose
user      35305  0.1  0.0 1679884 27280 ?       Sl   05:14   0:03 /usr/sbin/libvirtd --timeout=30
user      58953  0.0  0.0  11788  1488 ?        S    05:33   0:00 /bin/sh -c cd cockpit/; set -ex; PRIORITY=0009 TEST_NAME='pull-7423-20170802-053333' TEST_REVISION='a42f046b424c1cb4ec73673eb380440d25f1225a' bots/tests-invoke --rebase='master' 'verify/continuous-atomic' 'pull/7423/head'
user      58954  0.0  0.0 100156 11328 ?        S    05:33   0:00 python bots/tests-invoke --rebase=master verify/continuous-atomic pull/7423/head
user      58980  0.0  0.0  74732  3940 ?        S    05:33   0:00 ssh sink -- python sink pull-7423-20170802-053333-a42f046b-verify-continuous-atomic
user      71201  0.0  0.0 196600  4976 ?        Sl   05:36   0:00 /usr/sbin/virtlogd --timeout=30
user      71372  0.0  0.0   8624   696 ?        S    05:42   0:00 timeout 120m /build/cockpit/bots/../test/verify/run-tests --jobs 8
user      71373  0.0  0.0 254972 19548 ?        S    05:42   0:00 python /build/cockpit/bots/../test/verify/run-tests --jobs 8
user      73766  0.0  0.0 257092 14708 ?        S    05:46   0:00 python /build/cockpit/bots/../test/verify/run-tests --jobs 8
user      73774  8.9  0.5 2108740 665028 ?      Sl   05:46   1:44 /usr/bin/qemu-system-x86_64 -machine accel=kvm -name guest=continuous-atomic-127.0.0.2-2901,debug-threads=on -S -object secret,id=masterKey0,format=raw,file=/home/user/.config/libvirt/qemu/lib/domain-192-continuous-atomic-12/master-key.aes -machine pc-i440fx-2.7,accel=kvm,usb=off -cpu host -m 1024 -realtime mlock=off -smp 1,sockets=1,cores=1,threads=1 -uuid b12912b3-b90b-4e7c-b6a0-7070b5a4f105 -display none -no-user-config -nodefaults -chardev socket,id=charmonitor,path=/home/user/.config/libvirt/qemu/lib/domain-192-continuous-atomic-12/monitor.sock,server,nowait -mon chardev=charmonitor,id=monitor,mode=control -rtc base=utc -no-shutdown -boot strict=on -device piix3-usb-uhci,id=usb,bus=pci.0,addr=0x1.0x2 -device virtio-scsi-pci,id=scsi0,bus=pci.0,addr=0x2 -drive file=/build/tmp/run/5hOtxF.qcow2,format=qcow2,if=none,id=drive-virtio-disk0,serial=ROOT -device virtio-blk-pci,scsi=off,bus=pci.0,addr=0x3,drive=drive-virtio-disk0,id=virtio-disk0,bootindex=1 -drive file=/build/cockpit/test/common/cloud-init.iso,format=raw,if=none,id=drive-ide0-0-1,readonly=on -device ide-cd,bus=ide.0,unit=1,drive=drive-ide0-0-1,id=ide0-0-1 -chardev pty,id=charserial0 -device isa-serial,chardev=charserial0,id=serial0 -device virtio-balloon-pci,id=balloon0,bus=pci.0,addr=0x4 -netdev socket,mcast=230.0.0.1:6200,id=mcast0 -device rtl8139,netdev=mcast0,mac=52:54:01:00:00:01,bus=pci.0,addr=0x06 -netdev user,id=base0,restrict=on,net=172.27.0.0/24,hostname=m1.cockpit.lan,hostfwd=tcp:127.0.0.2:9791-:9090,hostfwd=tcp:127.0.0.2:2901-:22 -device rtl8139,netdev=base0,bus=pci.0,addr=0x05 -msg timestamp=on
user      74109  0.0  0.0  76848  4340 ?        S    05:46   0:00 ssh -p 2901 -i /build/cockpit/test/common/identity -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o BatchMode=yes -M -o ControlPath=/tmp/ssh-%h-%p-%r-73766 -o LogLevel=ERROR -l root 127.0.0.2 /bin/bash -c 'echo READY; read a'
user      74166  0.0  0.0  72516  3364 ?        S    05:46   0:00 ssh -p 2901 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o BatchMode=yes -o ControlPath=/tmp/ssh-%h-%p-%r-73766 -l root 127.0.0.2 systemctl start default.target
user      77595  0.0  0.0 257092 14904 ?        S    05:52   0:00 python /build/cockpit/bots/../test/verify/run-tests --jobs 8
user      82680  0.0  0.0  11920  1764 pts/0    Ss+  06:04   0:00 /bin/sh -i
user      82858  105  0.5 2521652 664812 ?      Sl   06:05   0:29 /usr/bin/qemu-system-x86_64 -machine accel=kvm -name guest=continuous-atomic-127.0.0.2-2201,debug-threads=on -S -object secret,id=masterKey0,format=raw,file=/home/user/.config/libvirt/qemu/lib/domain-250-continuous-atomic-12/master-key.aes -machine pc-i440fx-2.7,accel=kvm,usb=off -cpu host -m 1024 -realtime mlock=off -smp 1,sockets=1,cores=1,threads=1 -uuid ca29a9c3-c683-4569-a19a-f3424833ad23 -display none -no-user-config -nodefaults -chardev socket,id=charmonitor,path=/home/user/.config/libvirt/qemu/lib/domain-250-continuous-atomic-12/monitor.sock,server,nowait -mon chardev=charmonitor,id=monitor,mode=control -rtc base=utc -no-shutdown -boot strict=on -device piix3-usb-uhci,id=usb,bus=pci.0,addr=0x1.0x2 -device virtio-scsi-pci,id=scsi0,bus=pci.0,addr=0x2 -drive file=/build/tmp/run/9ZMgb4.qcow2,format=qcow2,if=none,id=drive-virtio-disk0,serial=ROOT -device virtio-blk-pci,scsi=off,bus=pci.0,addr=0x3,drive=drive-virtio-disk0,id=virtio-disk0,bootindex=1 -drive file=/build/cockpit/test/common/cloud-init.iso,format=raw,if=none,id=drive-ide0-0-1,readonly=on -device ide-cd,bus=ide.0,unit=1,drive=drive-ide0-0-1,id=ide0-0-1 -chardev pty,id=charserial0 -device isa-serial,chardev=charserial0,id=serial0 -device virtio-balloon-pci,id=balloon0,bus=pci.0,addr=0x4 -netdev socket,mcast=230.0.0.1:5500,id=mcast0 -device rtl8139,netdev=mcast0,mac=52:54:01:00:00:01,bus=pci.0,addr=0x06 -netdev user,id=base0,restrict=on,net=172.27.0.0/24,hostname=m1.cockpit.lan,hostfwd=tcp:127.0.0.2:9091-:9090,hostfwd=tcp:127.0.0.2:2201-:22 -device rtl8139,netdev=base0,bus=pci.0,addr=0x05 -msg timestamp=on
user      82938  0.0  0.0  11920  1788 ?        Ss   06:05   0:00 /bin/sh
user      82944  0.0  0.0  72516  3768 ?        S    06:05   0:00 ssh -p 2201 -i /build/cockpit/test/common/identity -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o BatchMode=yes -M -o ControlPath=/tmp/ssh-%h-%p-%r-77595 -o LogLevel=ERROR -l root 127.0.0.2 /bin/bash -c 'echo READY; read a'
```
This explains the hang: There are still two running QEMU processes for the two virsh instances, but there is no actual test running. The only python processes are the `tests-invoke` and `run-tests` wrappers and the sink.

There is a zombie ssh process, so possibly the actual `check-*` process crashed and left its ssh behind.

In the log, the last successful test that I see is
```
# ----------------------------------------------------------------------
# testTls (check_connection.TestConnection)
#
Warning: Permanently added '[127.0.0.2]:2201' (ECDSA) to the list of known hosts.
ok 147 testTls (check_connection.TestConnection) # duration: 50s
```
But of course that is not very predictable due to running 8 checks in parallel.

**Comment by martinpitt** [comment]
.. and now the test run actually did make some progress, the second virsh instance and the ssh zombie  are gone.  Looking at the individual test runtimes on the logs it just seems that most tests take two minutes or longer, while they usually take a magnitude of 20s for other images.

It seems we have a too long hang while waiting for the cockpit/js container to start up:
```
Warning: Permanently added '[127.0.0.2]:2701' (ECDSA) to the list of known hosts.
Resource Error: Connection closed http://127.0.0.2:9591/playground/pkgs
Restarting browser...
```
In between the first ssh connection and the `Resource Error` there is a long delay, even though the container already works (I can access it from the browser). So our polling for container startup seems broken, I look into that.

**Comment by martinpitt** [comment]
Fixed in PR #7434.

**Comment by martinpitt** [comment]
#7434 doesn't actually help. While the ResourceErrors are gone and individual tests run faster, collectively they don't as that minute of waiting is not using many resources, thus other tests just use CPU/IO during that time.

**Comment by martinpitt** [comment]
Reopening, #7434 did not help, it's something else.

**Comment by martinpitt** [comment]
The running test in #7434 is on cockpit-11, so it not only happens in our Openshift instances. Like in the original description above there is one virsh/qemu instance and these two commands being stuck for 1.5 hours:
```
user      18301  0.0  0.0  76844  4340 ?        S    04:03   0:00 ssh -p 2401 -i /build/cockpit/test/common/identity -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o BatchMode=yes -M -o ControlPath=/tmp/ssh-%h-%p-%r-18005 -o LogLevel=ERROR -l root 127.0.0.2 /bin/bash -c 'echo READY; read a'
user      18325  0.0  0.0  72512  3236 ?        S    04:03   0:00 ssh -p 2401 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o BatchMode=yes -o ControlPath=/tmp/ssh-%h-%p-%r-18005 -l root 127.0.0.2 systemctl start default.target
```

The first is from testvm.py's `_start_ssh_master()`, the second one is from `test/verify/check-services` - that finally identifies the hanging test \o/.

I can ssh into the VM and I see that `cloud-final.service` is running due to waiting for the network, just like I [previously debugged](https://github.com/cockpit-project/cockpit/pull/5886#issuecomment-318892992). So we really must fix that cloud-init/networking issue.

**Comment by martinpitt** [comment]
That ifconfig/ping cloud-init command used to be in test/common/make-cloud-init-iso, but #6334 removed it (that's why it didn't turn up in my grep). So we really just need to refresh the image, re-closing as duplicate of #7232.

**Comment by petervo** [comment]
Image has been rebuilt by #7531
